tests/integration/sriov_test.py

```diff
@@ -29,10 +29,10 @@
 
 
 @pytest.mark.xfail(
-    raises=NmstateNotImplementedError,
-    reason='SR-IOV is not supported yet',
+    raises=NotSupportedByDevice,
+    reason='The device does not support SR-IOV.',
     strict=True,
```

Comment:
IMHO it should not test eth1 but a dummy interface for the not-supported error and there should be an test that is allowed to fail (not with strict) so it could be tested on bare-metal where eth1 supports SR-IOV.

Reply:
We cannot use dummy interface to arise this error. We need it to be ethernet type. I like the idea of removing the "strict". What do you think? Thanks!

Reply:
Currently the test is called `...not_supported`. To keep this name, it should ideally always fail, regardless of the hardware. An alternative to the dummy would be a veth interface or you could mock the NM check to always tell that it is not supported. Not sure if this is too much effort, though.

Whats missing then are tests for the actual features.

Reply:
Missing tests for the actual features added.